pkg/generate/app/imagestreamlookup.go

```diff
@@ -56,10 +58,20 @@ func (r ImageStreamSearcher) Search(terms ...string) (ComponentMatches, error) {
 				} else {
 					imageref, _ := imageapi.ParseDockerImageReference(term)
 					imageref.Name = stream.Name
+
+					localSearchTag := searchTag
```

Comment:
near as i can tell the intent is "this is either the original searchTag provided by the user, or it is the Tag which is followed by the :latest tag", which seems hard to convey in a name, so maybe a comment would be better.